polaris-plugins/polaris-plugins-connector/connector-consul-api/src/main/java/com/tencent/polaris/plugins/connector/consul/ConsulAPIConnector.java

```diff
@@ -0,0 +1,250 @@
+ * <li>registerInstance/deregisterInstance to register/deregister an instance.
+ * <li>heartbeat to send heartbeat manually.
+ * </ol>
+ *
+ * @author Haotian Zhang
+ */
+public class ConsulAPIConnector extends DestroyableServerConnector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConsulAPIConnector.class);
+
+    /**
+     * If server connector initialized.
+     */
+    private boolean initialized = false;
+
+    private ConsulClient consulClient;
+
+    private String id;
+
+    @Override
+    public String getName() {
+        return DefaultPlugins.SERVER_CONNECTOR_CONSUL;
+    }
+
+    @Override
+    public PluginType getType() {
+        return PluginTypes.SERVER_CONNECTOR.getBaseType();
+    }
+
+    @Override
+    public void init(InitContext ctx) throws PolarisException {
+        if (!initialized) {
+            List<ServerConnectorConfigImpl> serverConnectorConfigs = ctx.getConfig().getGlobal().getServerConnectors();
+            if (CollectionUtils.isNotEmpty(serverConnectorConfigs)) {
+                for (ServerConnectorConfigImpl serverConnectorConfig : serverConnectorConfigs) {
+                    if (DefaultPlugins.SERVER_CONNECTOR_CONSUL.equals(serverConnectorConfig.getProtocol())) {
+                        initActually(ctx, serverConnectorConfig);
+                    }
+                }
+            }
+        }
+    }
+
+    private void initActually(InitContext ctx, ServerConnectorConfig connectorConfig) {
+        String address = connectorConfig.getAddresses().get(0);
+        String[] addressSplit = address.split(":");
+        String agentHost = addressSplit[0];
+        int agentPort = Integer.parseInt(addressSplit[1]);
+        consulClient = new ConsulClient(agentHost, agentPort);
+        initialized = true;
+    }
+
+    @Override
+    public void postContextInit(Extensions ctx) throws PolarisException {
+        // do nothing
+    }
+
+    @Override
+    public void registerServiceHandler(ServiceEventHandler handler) throws PolarisException {
+        // do nothing
+    }
+
+    @Override
+    public void deRegisterServiceHandler(ServiceEventKey eventKey) throws PolarisException {
+        // do nothing
+    }
+
+    @Override
+    public CommonProviderResponse registerInstance(CommonProviderRequest req) throws PolarisException {
+        LOG.info("Registering service to Consul");
+        NewService service = null;
+        try {
+            service = buildRegisterInstanceRequest(req);
+            this.consulClient.agentServiceRegister(service);
+            CommonProviderResponse resp = new CommonProviderResponse();
+            id = service.getId();
+            resp.setInstanceID(service.getId());
+            resp.setExists(true);
+            LOG.info("Registered service to Consul: " + service);
+            return resp;
+        } catch (ConsulException e) {
+            LOG.warn("Register instance to Consul failed of service: " + service, e);
+        }
+        return null;
```

Comment:
shoud wrap exception to PolarisException?  can refer to GrpcConnector .

And shoud not return null